srtcore/sync.cpp

```diff
@@ -446,3 +454,69 @@ int srt::sync::SyncEvent::wait_for_monotonic(pthread_cond_t* cond, pthread_mutex
     return wait_for(cond, mutex, rel_time);
 }
 #endif
+
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// CEvent class
+//
+////////////////////////////////////////////////////////////////////////////////
+
+srt::sync::CEvent::CEvent()
+{
+#ifndef _WIN32
+    m_cond.init();
+#endif
+}
+
+
+srt::sync::CEvent::~CEvent()
+{
+#ifndef _WIN32
+    m_cond.destroy();
+#endif
+}
+
+
+bool srt::sync::CEvent::wait_until(const TimePoint<steady_clock>& tp)
```

Comment:
Not sure if this isn't actually limiting any usefulness of this class. Many application cases lock the section for a large range and treat the wait call similarly as `InvertedLock` class. In such a code this function will be useless.

Maybe at least THIS case should be renamed to `lock_wait_until` etc.

Reply:
Given that we now have a number of redundant classes, I would postpone this decision once a common monotonic clock is done and C++ mappings are introduced.
There will have to be some cleaning stage anyway.

Reply:
At least this function that does locking by itself should not have the same name as the function that does waiting without locking. It will cause confusion.